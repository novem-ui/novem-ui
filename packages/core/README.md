# @novem-ui/core

This lib exports all the Novem UI components