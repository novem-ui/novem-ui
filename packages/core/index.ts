export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
