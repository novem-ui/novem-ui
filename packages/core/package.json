{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
