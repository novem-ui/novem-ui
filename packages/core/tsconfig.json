{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
