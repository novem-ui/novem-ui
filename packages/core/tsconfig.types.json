{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
