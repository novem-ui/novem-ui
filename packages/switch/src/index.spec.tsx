import React from 'react'

import { render } from '@testing-library/react'
import { NovemProvider } from '@novem-ui/core'

import Switch, { SwitchProps } from '.'

const setup = (props: SwitchProps = {}) => {
  const result = render(
    <NovemProvider>
      <Switch {...props} />
    </NovemProvider>
  )
  return result
}

describe('<Switch />', () => {
  test('should render a switch component', () => {
    const { container } = setup()
    expect(container).toMatchSnapshot()
  })

  test('should render a switch component (checked)', () => {
    const { container } = setup({ checked: true })
    expect(container).toMatchSnapshot()
  })

  test('should render a switch component (custom color)', () => {
    const { container } = setup({ baseColor: 'red' })
    expect(container).toMatchSnapshot()
  })

  test('should render a badge component (disabled)', () => {
    const { container } = setup({ disabled: true })
    expect(container).toMatchSnapshot()
  })
})
