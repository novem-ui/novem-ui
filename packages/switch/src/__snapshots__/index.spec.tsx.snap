// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render a badge component (checked) 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  width: 2rem;
  height: 1.25rem;
}

.emotion-2 {
  opacity: 0;
  width: 0;
  height: 0;
}

.emotion-2:checked+.emotion-5::after {
  -webkit-transform: translate3d(12px, 0, 0);
  -moz-transform: translate3d(12px, 0, 0);
  -ms-transform: translate3d(12px, 0, 0);
  transform: translate3d(12px, 0, 0);
}

.emotion-4 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  bottom: 0;
  cursor: pointer;
  left: 0;
  right: 0;
  box-sizing: border-box;
  top: 0;
  -webkit-transition: all ease-in 0.25s;
  transition: all ease-in 0.25s;
  position: absolute;
  border-radius: 0.625rem;
  padding: 0.125rem;
}

.emotion-4::after {
  content: '';
  background-color: white;
  border-radius: 0.5rem;
  display: block;
  height: 1rem;
  -webkit-transition: -webkit-transform ease 0.25s;
  transition: transform ease 0.25s;
  width: 1rem;
  box-shadow: 0px 1px 2px 2px rgba(51,50,50,0.1);
}

<div>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      checked=""
      class="emotion-2 emotion-3"
      type="checkbox"
    />
    <span
      class="emotion-4 emotion-5"
    />
  </label>
</div>
`;

exports[`<Badge /> should render a badge component (custom color) 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  width: 2rem;
  height: 1.25rem;
}

.emotion-2 {
  opacity: 0;
  width: 0;
  height: 0;
}

.emotion-2:checked+.emotion-5::after {
  -webkit-transform: translate3d(12px, 0, 0);
  -moz-transform: translate3d(12px, 0, 0);
  -ms-transform: translate3d(12px, 0, 0);
  transform: translate3d(12px, 0, 0);
}

.emotion-4 {
  background-color: #EB5757;
  color: rgba(255,255,255,1);
  bottom: 0;
  cursor: pointer;
  left: 0;
  right: 0;
  box-sizing: border-box;
  top: 0;
  -webkit-transition: all ease-in 0.25s;
  transition: all ease-in 0.25s;
  position: absolute;
  border-radius: 0.625rem;
  padding: 0.125rem;
}

.emotion-4::after {
  content: '';
  background-color: white;
  border-radius: 0.5rem;
  display: block;
  height: 1rem;
  -webkit-transition: -webkit-transform ease 0.25s;
  transition: transform ease 0.25s;
  width: 1rem;
  box-shadow: 0px 1px 2px 2px rgba(51,50,50,0.1);
}

<div>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-3"
      type="checkbox"
    />
    <span
      class="emotion-4 emotion-5"
    />
  </label>
</div>
`;

exports[`<Badge /> should render a badge component 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  width: 2rem;
  height: 1.25rem;
}

.emotion-2 {
  opacity: 0;
  width: 0;
  height: 0;
}

.emotion-2:checked+.emotion-5::after {
  -webkit-transform: translate3d(12px, 0, 0);
  -moz-transform: translate3d(12px, 0, 0);
  -ms-transform: translate3d(12px, 0, 0);
  transform: translate3d(12px, 0, 0);
}

.emotion-4 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  bottom: 0;
  cursor: pointer;
  left: 0;
  right: 0;
  box-sizing: border-box;
  top: 0;
  -webkit-transition: all ease-in 0.25s;
  transition: all ease-in 0.25s;
  position: absolute;
  border-radius: 0.625rem;
  padding: 0.125rem;
}

.emotion-4::after {
  content: '';
  background-color: white;
  border-radius: 0.5rem;
  display: block;
  height: 1rem;
  -webkit-transition: -webkit-transform ease 0.25s;
  transition: transform ease 0.25s;
  width: 1rem;
  box-shadow: 0px 1px 2px 2px rgba(51,50,50,0.1);
}

<div>
  <label
    class="emotion-0 emotion-1"
  >
    <input
      class="emotion-2 emotion-3"
      type="checkbox"
    />
    <span
      class="emotion-4 emotion-5"
    />
  </label>
</div>
`;
